packages/logic-utils/src/utils/documentation.ts

```diff
@@ -0,0 +1,191 @@
+import {
+  UI5DeprecatedInfo,
+  UI5SemanticModel,
+} from "@ui5-language-assistant/semantic-model-types";
+import { ReplaceFunction, forEach, unescape } from "lodash";
+
+const MAX_SNIPPET_LENGTH = 500;
+
+export function getDeprecationPlainTextSnippet(
+  title: string | undefined,
+  deprecatedInfo: UI5DeprecatedInfo,
+  model: UI5SemanticModel
+): string {
+  let text = deprecatedInfo.text;
+  if (text !== undefined) {
+    text = convertJSDocToPlainText(text, model);
+    // Only take the first sentence.
+    // A sentence is defined by a dot followed by a whitespace character.
+    // TODO Should we allow the result to have more than 1 line? (If so, add "s" flag to the regex)
+    const matches = text.match(/^((?:.*?)\.)\s/);
+    if (matches !== null) {
+      text = matches[1];
+    }
+    // Limit the length of the snippet
+    if (text.length > MAX_SNIPPET_LENGTH) {
+      text = text.substring(0, MAX_SNIPPET_LENGTH) + "...";
```

Comment:
The "..." dots are not that usable when there is no way to open the link or expand them for the full text, but I do not have an alternative to that :(

Reply:
Since we have hover now, you will either see it in the description (if you hover on the problem marker) or if you see the message in the problem view, click on it and hover over the message. (That's why I really wanted to have the hover before we did this change.)

Reply:
Oh cool good point, this resolves the problem.